oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/PhysicalIndices.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.skywalking.oap.server.storage.plugin.elasticsearch.base;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class PhysicalIndices {
```

Comment:
Please explain differences and boundaries between `PhysicalIndexManager` and `PhysicalIndices`.
Right now, PhysicalIndexManager becomes a util, as it doesn't include any new fields. These two should be merged?

Reply:
I am thing about
```
public enum IndexController {
   ....
   public class LogicIndicesRegister {
        private static final Map<String, String> LOGIC_INDECES_CATALOG = new HashMap<>();
   }
}
```
How about this?

Reply:
I think your latest change still separating `PhysicalIndexManager and PhysicalIndices` as `PhysicalIndices` and `PhysicalIndexer`. Could you explain why?
And again on coding style, you have `PhysicalIndexer` having all instance level methods, but actually no instance field, and at the same time, you have PhysicalIndices with a static field, with all static methods. These 2 typically should be reverted. No instance method means no point to maintain the instance. Why?

Please read my above suggestion. If you want a enum to provide an API set, it is fine, but please keep consistent.

Reply:
Right now, I am thinking about this
```
public enum IndexController {
   // Move methods in PhysicalIndexer here
   ....
   public class LogicIndicesRegister {
        // Move PhysicalIndices methods into here
        private final Map<String, String> LOGIC_INDECES_CATALOG = new HashMap<>();
   }

  public class Checker {
       // Check the template and index compatible 
       // Note, the compatible check require column name, type, mapping, analyzer, etc. are all same
  }
}
```